napari/_qt/qt_main_window.py

```diff
@@ -1271,12 +1273,8 @@ def _update_theme(self, event=None):
             self._qt_viewer.viewer.theme = value
             if value == "system":
                 # system isn't a theme, so get the name and set style sheet
-                actual_theme_name = get_system_theme()
-                self._qt_window.setStyleSheet(
-                    get_stylesheet(actual_theme_name)
-                )
-            else:
-                self._qt_window.setStyleSheet(get_stylesheet(value))
+                value = get_system_theme()
+            self._qt_window.setStyleSheet(get_stylesheet(value))
```

Comment:
+1

Reply:
So this changes the theme behavior, such that viewer.theme only returns `light` or `dark`.
I'm actually supportive of this, but when I implemented the disconnect between the settings and the current viewer (#5143), it was contentious, with the consensus being that viewer.theme should report `system` if that was indeed the theme set and that the stylesheet should be adjusted appropriately. By letting viewer.theme retain `system`, it may help future serialization of the viewer state or let the theme adapt to the OS theme.
See these two comments:
https://github.com/napari/napari/pull/5143#issuecomment-1336213640
https://github.com/napari/napari/pull/5143#issuecomment-1342178406

I'm not sure it's needed to change this behavior to implement the changes.

Reply:
Checking this change, I think the value of `viewer.theme` still can be `system`. Seems to me that the change here is just a little refactor to not have to call twice the `self._qt_window.setStyleSheet` while reusing the `value` variable, no? Testing locally, if I selected `system` as theme value I can still see from the IPython Console that `viewer.theme` outputs `system`